denonavr/denonavr.py

```diff
@@ -1865,6 +1915,103 @@ def mute(self, mute):
             _LOGGER.error("Connection error: mute command not sent.")
             return False
 
+    def _set_tone_control_command(self, parameter_type, value):
+        """Post request for tone control commands."""
+        root = ET.Element("tx")
+        ET.SubElement(root, 'cmd', id='1').text = 'SetToneControl'
+        ET.SubElement(root, parameter_type).text = str(value)
+        tree = ET.ElementTree(root)
+
+        body = BytesIO()
+        tree.write(body, encoding="utf-8", xml_declaration=True)
+
+        try:
+            self.send_post_command(
+                command=self._urls.appcommand, body=body.getvalue()
+                )
+        except (
+                requests.exceptions.ConnectTimeout,
+                requests.exceptions.RequestException
+                ):
+            _LOGGER.error("No connection to %s end point on host %s",
+                          self._urls.appcommand, self._host)
+            return False
+        return True
+
+    def enable_tone_control(self):
+        """Enable tone control to change settings like bass or treble."""
+        if self._tone_control_adjust is True:
+            return True
+        elif self._set_tone_control_command(parameter_type='adjust', value=1):
+            self._tone_control_adjust = True
+            return True
```

Comment:
This check does not cover the case, when tone control cannot be enabled because Dynamic EQ is active.
A way to check this could be to run `self._set_tone_control_command` first, then run self._update_tone_control again and check if `self._tone_control_adjust` is really True